src/node_crypto.cc

```diff
@@ -1681,10 +1697,49 @@ static Local<Object> X509ToObject(Environment* env, X509* cert) {
         reinterpret_cast<unsigned char*>(Buffer::Data(pubbuff));
     i2d_RSA_PUBKEY(rsa.get(), &pubserialized);
     info->Set(env->context(), env->pubkey_string(), pubbuff).FromJust();
+  } else if (ec) {
+    const EC_GROUP* group = EC_KEY_get0_group(ec.get());
+    if (group) {
+      int ecbits = EC_GROUP_order_bits(group);
+      if (ecbits > 0) {
+        info->Set(context, env->ecbits_string(),
+                  Integer::New(env->isolate(), ecbits)).FromJust();
+      }
+    }
+
+    unsigned char* pub = nullptr;
+    size_t publen = EC_KEY_key2buf(ec.get(), EC_KEY_get_conv_form(ec.get()),
+                                   &pub, nullptr);
+    if (publen) {
+      Local<Object> buf = Buffer::New(env, pub, publen).ToLocalChecked();
+      info->Set(context, env->ecpubkey_string(), buf).FromJust();
+    } else {
+      CHECK(pub == nullptr);
+    }
+
+    if (EC_GROUP_get_asn1_flag(group)) {
+      // Curve is well-known, get its OID and NIST nick-name (if it has one).
+
+      int nid = EC_GROUP_get_curve_name(group);
+      const char* sn = nullptr;
+      if (nid && (sn = OBJ_nid2sn(nid))) {
```

Comment:
The style guide doesn't strictly forbid assignment in comparison, I think, but it's really unidiomatic for the code base.